### PR TITLE
Fixed broken flaky front-end tests, not their flakiness

### PR DESCRIPTION
**Pull request by Theodlz**
This PR aims to fix broken flaky front-end tests. It seems that as those are flaky, after the frontend was modified, the fact that they failed seemed "normal" due to their flakiness, when in reality, they are simply broken.

Current state: All of them pass locally.

Edit 1: sadly, even if they do pass locally (they are not broken anymore), most of them don't pass when running the whole test suite here on git. I removed the rerun so to directly and quickly see if they are flaky or not.

### CODE REVIEW

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 from regions import Regions
 
 from skyportal.tests import api
 from selenium.webdriver.common.keys import Keys
+from selenium.common.exceptions import TimeoutException
+from datetime import date, timedelta
 
 
-@pytest.mark.flaky(reruns=2)
+# @pytest.mark.flaky(reruns=2)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:13]: We can just get rid of these if solved?

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 2]
```diff
@@ -69,7 +70,7 @@ def test_telescope_frontend_desktop(super_admin_token, super_admin_user, driver)
     driver.wait_for_xpath(f'//*[@id="{name2}_info"]')
 
 
-@pytest.mark.flaky(reruns=2)
+# @pytest.mark.flaky(reruns=2)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:73]: Same here.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 3]
```diff
@@ -116,8 +117,11 @@ def test_telescope_frontend_mobile(super_admin_token, super_admin_user, driver):
     # check for dropdown instrument
     driver.wait_for_xpath(f'//span[text()="{name2}"]')
 
+    driver.set_window_position(0, 0)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:120]: Ah this is a good trick.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 4]
```diff
@@ -199,7 +206,7 @@ def test_instrument_frontend(super_admin_token, super_admin_user, driver):
     driver.wait_for_xpath('//span[contains(text(), "Instrument name matches another")]')
 
 
-@pytest.mark.flaky(reruns=3)
+# @pytest.mark.flaky(reruns=3)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:209]: Same here.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 5]
```diff
@@ -447,13 +455,13 @@ def test_gcnevents_observations(
     driver.click_xpath(submit_button_xpath)
 
     # check that the executed observation table appears
-    driver.wait_for_xpath('//*[text()="84434604"]')
+    driver.wait_for_xpath('//*[text()="84434604"]', timeout=10)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:458]: Ah yes.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 6]
```diff
@@ -447,13 +455,13 @@ def test_gcnevents_observations(
...
     driver.wait_for_xpath('//*[text()="ztfr"]')
     driver.wait_for_xpath('//*[text()="1.57415"]')
     driver.wait_for_xpath('//*[text()="20.40705"]')
 
 
-@pytest.mark.flaky(reruns=2)
+# @pytest.mark.flaky(reruns=2)
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:464]: Same here.

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 7]
```diff
@@ -814,15 +830,24 @@ def test_gcn_request(driver, user, super_admin_token, public_group):
...
-    select_box.click()
+    # adding a retry loop because the list of instruments can take time to be fetched
+    driver.wait_for_xpath('//*[@id="instrumentSelectLabel"]/../*/*[@role="button"]')
+    driver.click_xpath('//*[@id="instrumentSelectLabel"]/../*/*[@role="button"]')
 
-    driver.click_xpath(
-        f'//li[contains(text(), "{telescope_name}")][contains(text(), "{instrument_name}")]',
-        scroll_parent=True,
-    )
+    # driver.click_xpath(
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:837]: Can we just delete?

**Review by mcoughlin — APPROVED** [review]
Looks great @Theodlz. I think can just remove the commented things.

**Comment by Theodlz** [comment]
Hey @mcoughlin, sadly as i said in the edit, even if they pass locally now and aren't broken anymore, they are still flaky and don't work well in the GitHub Actions. This PR is a work in progress and it needs way more work, which is why i left the `rerun` commented and didn remove them, as it is not solved yet.

**Comment by profjsb** [comment]
What more is needed on this PR @Theodlz ?

**Comment by Theodlz** [comment]
> What more is needed on this PR @Theodlz ?

Hi @profjsb. All the tests are fixed (they can pass, but still flaky), so i guess this PR did its job. I'm still running them multiple times to see where the flakiness comes from, but this could be the subject of another PR later on : `Flaky Frontend tests - Solve Flakiness` ?

File: skyportal/tests/flaky/test_frontend_flaky.py [hunk 8]
```diff
@@ -814,15 +850,24 @@ def test_gcn_request(driver, user, super_admin_token, public_group):
...
```
Comment (mcoughlin) [inline skyportal/tests/flaky/test_frontend_flaky.py:857]: You can just delete this part yes?

**Review by mcoughlin — APPROVED** [review]
I would just delete any commented sections now that it is fixed

**Review by mcoughlin — APPROVED** [review]
Thanks @Theodlz. Feel free to merge when tests pass.

**Comment by Theodlz** [comment]
@mcoughlin well since they are flaky that will probably never happen though hahaha. I will try putting them in a non flaky folder just to see it gets better this way.

**Comment by mcoughlin** [comment]
Ah yes, good point @Theodlz. Anyways, when the green check marks are there, should be good to put in. Thanks for fixing those.

**Pull request merged**